server_test.go

```diff
@@ -0,0 +1,179 @@
+package timetracker_test
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+	"timetracker"
+)
+
+func TestGetEnvironmentVariables(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		fn func(string) (string, error)
+		a  string
+	}
+	tcs := []testCase{
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_HOST"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_PORT"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_USER"},
+		{fn: timetracker.GetEnvironmentVariable, a: "TIMETRACKER_DB_NAME"},
+	}
+	for _, tc := range tcs {
+		_, err := tc.fn(tc.a)
+		if err != nil {
+			t.Fatalf("error with environment variable: %s %s", tc.a, err)
+		}
+	}
+
+}
+
+func TestServerRoutesStatusOK(t *testing.T) {
+	t.Parallel()
+
+	type testCase struct {
+		route string
+	}
+	tcs := []testCase{
+		{route: "http://127.0.0.1:9000"},
+		{route: "http://127.0.0.1:9000/task/report"},
+		{route: "http://127.0.0.1:9000/task/create"},
+	}
+	s := timetracker.NewServer(
+		timetracker.WithPort(9000),
+		timetracker.WithLogLevel("quiet"),
+	)
+
+	go func() {
+		err := s.ListenAndServe()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	for _, tc := range tcs {
+		resp, err := http.Get(tc.route)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode != http.StatusOK {
+			t.Fatalf("unexpected status %d", resp.StatusCode)
+		}
+
+	}
+
+}
+
+func TestRenderHomePage(t *testing.T) {
+	t.Parallel()
+
+	tasks := []timetracker.Task{
+		{
+			Name:        "piano",
+			StartTime:   time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC),
```

Comment:
It's usually easier to read the test data if it's in the form of a call to `time.Parse`:

```go
StartTime: time.Parse(time.RFC3339, "2021-01-01T00:00:00+00:00"),
```

Reply:
ah, nice formatting